framework/src/modules/chain/helpers/jobs_queue.js

```diff
@@ -51,17 +51,29 @@ const jobsQueue = {
 		// Check if job is function, name is string and time is integer
 		if (
 			!job ||
-			Object.prototype.toString.call(job) !== '[object Function]' ||
+			!(
+				Object.prototype.toString.call(job) === '[object Function]' ||
+				Object.prototype.toString.call(job) === '[object AsyncFunction]'
```

Comment:
I don't prefer using `prototype` functions. You can use alternative approach `job.constructor.name === 'Function'`

Reply:
I prefer `typeof job === 'function'` since `job.constructor.name` can be tampered with if someone (or a library) messes with the prototype chain.